src/rpc/core_rpc_server.cpp

```diff
@@ -2655,6 +2655,8 @@ namespace cryptonote
     entry.last_reward_transaction_index = sn_info.info.last_reward_transaction_index;
     entry.last_uptime_proof             = sn_info.info.proof.timestamp;
     entry.active                        = sn_info.info.is_active();
+    entry.state_height                  = sn_info.info.is_fully_funded()
```

Comment:
Active since height is negative right if decommissioned? This is u64, I'm thinking you want active to instead be a state string, "active from height", "decommissioned at height",  "not funded (last reward height)"

Reply:
Active since height will only be negative if the state is decommissioned, but in that case this gets set to `sn_info.info.last_decommission_height` instead of `active_since_height`.

Reply:
As for state strings: this is a JSON interface, not a human consumption interface.  JSON consumers really shouldn't have to look for human-readable strings.  It could include multiple bools (`active`, `decommissioned`, `funded`) but those seem redundant as the data isn't that complicated to extract currently: `active` is provided, `funded = contributions > required`, and `decommissioned = funded && !active`.

Reply:
The alternative is to document this in the comments. It's not immediately clear to someone trying to use this data if you have to go into the source code to figure out in which case this value represents 1 of 3 different values. The 3 bools vs 1 string, i definitely prefer a string rather than the 1 bool if true, invalidates the 2 other bools of the response.

Reply:
How about `active` and `funded` bools?  Then:
`active` means, well, active (funded and not decomm'ed)
`funded` means has funding.
`funded && !active` means decomm'ed.

Reply:
Sound fine